pkg/injection/codemodule/installer/job/installer.go

```diff
@@ -0,0 +1,121 @@
+package job
+
+import (
+	"context"
+	"os"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/csi/metadata"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/common"
+	"github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/symlink"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/env"
+	jobutil "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/job"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Properties struct {
+	Owner        client.Object
+	ApiReader    client.Reader
+	Client       client.Client
+	ImageUri     string
+	PathResolver metadata.PathResolver
+	PullSecrets  []string
+}
+
+func NewInstaller(ctx context.Context, fs afero.Fs, props *Properties) installer.Installer {
+	return &Installer{
+		fs:       fs,
+		props:    props,
+		nodeName: env.GetNodeName(),
+	}
+}
+
+type Installer struct {
+	fs       afero.Fs
+	props    *Properties
+	nodeName string
+}
+
+func (inst *Installer) InstallAgent(ctx context.Context, targetDir string) (bool, error) {
```

Comment:
```suggestion
func (i *Installer) InstallAgent(ctx context.Context, targetDir string) (bool, error) {
```
there is no strict rule :) but short name what we do everywhere (if you don't want to confuse with `i` what about `in` ?
it's not strict suggestion just more like question.

Reply:
`i` is "iterator" so nonono

`in` is (to me) a python keyword :D

`inst` is to shortest I liked 😅